src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.Reflection.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System;
+using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static partial class ActivatorUtilities
+    {
+#if NET8_0_OR_GREATER // Use the faster ConstructorInvoker which also has alloc-free APIs when <= 4 parameters.
+        private static object ReflectionFactory_ServiceOnly_Fixed(
+            ConstructorInvoker invoker,
+            Type[] parameterTypes,
+            Type declaringType,
+            IServiceProvider serviceProvider)
+        {
+            Debug.Assert(parameterTypes.Length >= 1 && parameterTypes.Length <= FixedArgumentThreshold);
+            Debug.Assert(FixedArgumentThreshold == 4);
+
+            if (serviceProvider is null)
```

Comment:
Why isn't this written the same as [line 56](https://github.com/dotnet/runtime/pull/89573/files#diff-c055ce0926512dd28ea3c7e1471196e504404bc5f448cb9a0a7e606f420f06feR56)?

```csharp
ArgumentNullException.ThrowIfNull(nameof(serviceProvider));
```

Reply:
See my other comment below on this.